java/code/src/com/suse/manager/matcher/test/MatcherJsonIOTest.java

```diff
@@ -65,19 +68,22 @@
     private static final String AMD64_ARCH = "amd64";
     private static final String S390_ARCH = "s390";
     private static final String PPC64LE_ARCH = "ppc64le";
+    private SystemEntitlementManager systemEntitlementManager;
 
     @Override
     public void setUp() throws Exception {
         super.setUp();
         setImposteriser(ClassImposteriser.INSTANCE);
 
-        SaltService saltServiceMock = context().mock(SaltService.class);
-        context().checking(new Expectations() {{
-            allowing(saltServiceMock).callSync(
-                    with(any(LocalCall.class)),
-                    with(any(String.class)));
-        }});
-        SystemEntitler.INSTANCE.setSaltService(saltServiceMock);
+        SaltService saltServiceMock = new SaltService() {
```

Comment:
The primary reason for getting rid of the mocking framework is to make tests easier to understand/write since everybody knows basic inheritance and overriding methods.
Another minor benefit is failure at compile time rather then runtime when changing return types of methods. 
I don't worry about calls going undetected since that would mean the test passing. It could be more difficult to see why something fails in some case in most it will probably lead to an exception with a stacktrace since most methods are mocking things that are not available in the test environment. We could retain this behavior by having a test implementation of each interface that throws an not implemented exception for every method that is not overridden.

Reply:
I agree with this perspective and leave it up to you to decide if and when such a test implementation makes sense. Thanks